rust/common/error.rs

```diff
@@ -20,25 +20,32 @@
  *
  */
 
-use crate::write_joined;
+use std::fmt;
+
+use chrono::NaiveDateTime;
 use pest::error::{Error as PestError, LineColLocation};
-use std::{convert::Infallible, fmt};
+
+use crate::{
+    common::token,
+    pattern::{Label, Pattern, Reference, ThingVariable, TypeVariable, UnboundVariable, Value},
+    write_joined,
+};
```

Comment:
This arrangement of imports, dubbed StdExternalCrate, is a de facto Rust community standard. Enforced here to comply with client-rust formatting, where it makes a much bigger difference.